salesforce_api/kafka_salesforce_integration/kafka-salesforce-pricebook_update/main.bal

```diff
@@ -0,0 +1,49 @@
+import ballerinax/kafka;
+import ballerinax/salesforce as sfdc;
+
+const string PRICEBOOKID = "";
+configurable SalesforceOAuth2Config salesforceOAuthConfig = ?;
+configurable string salesforceBaseUrl = ?;
+public type ProductPrice readonly & record {
+    string Name;
+    float UnitPrice;
+};
+
+public type ProductPriceUpdate readonly & record {
+    float UnitPrice;
+};
+
+listener kafka:Listener orderListener = new (kafka:DEFAULT_URL, {
+    groupId: "order-group-id",
+    topics: "foobar"
+});
+
+type SalesforceOAuth2Config record {
+    string clientId;
+    string clientSecret;
+    string refreshToken;
+    string refreshUrl = "https://test.salesforce.com/services/oauth2/token";
+};
+
+// Salesforce client
+sfdc:Client sfdcClient = check new ({
+    baseUrl: salesforceBaseUrl,
+    auth: {
+        clientId: salesforceOAuthConfig.clientId,
+        clientSecret: salesforceOAuthConfig.clientSecret,
+        refreshToken: salesforceOAuthConfig.refreshToken,
+        refreshUrl: salesforceOAuthConfig.refreshUrl
+    }
+});
+
+service on orderListener {
+    remote function onConsumerRecord(ProductPrice[] prices) returns error? {
+        foreach ProductPrice price in prices {
```

Comment:
```suggestion
        foreach ProductPrice {name, unitPrice} in prices {
```

and use `name` and `unitPrice` directly in field access.

Reply:
Updated in [dce4555](https://github.com/ballerina-guides/integration-samples/pull/31/commits/dce4555bea3a940121206d5b82b8d22c3aa4366c)